.github/workflows/lint-and-unit-test.yml

```diff
@@ -0,0 +1,42 @@
+name: Lint and unit test
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '**/**.go'
+  pull_request:
+    paths:
+      - '**/**.go'
+
+jobs:
+  lint-and-unit-test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+    
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.21
+    
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo add-apt-repository universe && sudo apt-get install -y libswscale-dev libavcodec-dev libavformat-dev libavutil-dev ffmpeg
+
+    - name: Check gofmt
+      run: |
+        gofmt -d -s .
+        if [ -n "$(gofmt -d -s .)" ]; then
+          echo "gofmt is unhappy with your code quality!! Please run 'make lint' and commit the changes."
+          exit 1
+        fi
```

Comment:
Is this the check that RDK does? I think this should:
1. Run gofmt recursively (idk if this command does that or not)
2. Check if any files tracked by git changed
3. If yes, fail the build

I think this will only fail the build if the code is unlintable, not if the code was not linted before being committed which is what we want to check for.

Reply:
`gofmt -d -s .` we can take out honestly now that I re-read it since `-n "$(gofmt -d -s .)"` should be sufficient.

The flags: `-d` outputs a diff, `-s` just simplifies it (we can take it out too)

`-n "$(gofmt -d -s .)"` `-n` is true if the following command is non-empty

[gofmt commands are recursive](https://pkg.go.dev/cmd/gofmt) within the specified dir

Overall it should do exactly those 1, 2, 3 steps you outlined... except for maybe the tracked by git part, but I can't really think of a situation where .go files in the repo aren't tracked

Reply:
Sounds good, thanks!